spec/features/commands_spec.rb

```diff
@@ -11,6 +11,16 @@
     HEAD('acme-inc/api', 'master',  'ad80a1b3e1a94b98ce99b71a48f811f1')
     HEAD('acme-inc/api', 'topic',   '4c7b474c6e1c81553a16d1082cebfa60')
     HEAD('acme-inc/api', 'failing', '46c2acc4e588924340adcd108cfc948b')
+
+    stub_request(:get, "https://api.github.com/repos/acme-inc/api").
+      with(
+        headers: {
+        'Accept'=>'application/vnd.github.v3+json',
+        'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+        'Content-Type'=>'application/json',
+        'User-Agent'=>'Octokit Ruby Gem 4.16.0'
+        }).
+      to_return(status: 200, body: {'default_ref': 'main'}.to_json, headers: { 'Content-Type' => 'application/json' })
```

Comment:
The tests that depend on this response expect a `default_ref` to be defined in the body of this response. This is definitely something that I'm not completely understanding. We break a number of tests in a number of different ways if we don't stub this `default_ref`. If you set this to `default_branch` then we break the tests attached below. 
<img width="899" alt="Screen Shot 2021-12-13 at 10 47 29 AM" src="https://user-images.githubusercontent.com/23040094/145862489-c4854d0b-c576-4754-aaff-8c9c2a2ebbdf.png">

Reply:
Hm, weird. Maybe I'm misunderstanding but this is supposed to be a mock response from the GitHub API, no? If the tests are failing when it returns a response that looks like the real API response then I think there's something broken in the code, rather than something broken in the test mock.

Reply:
Yea, it's just that the tests are looking for `master` in most cases, but getting `main` instead. I can make some updates that will get it working.

Reply:
Because these tests depend on an external repo (acme-inc) that has a default_branch of `master` , we need to keep that expectation in the tests.

Reply:
I'm unclear on the resolution here. I still see some tests using default_ref and some using default_branch; the github docs don't list default_ref as a response param anywhere. 

We need to understand this before changing things here.

Reply:
Apologies for the confusion, I needed to update the environment_spec stubbed request as well. `default_ref` is not a response param we should expect, but `default_branch` is. All the tests now reflect that. If we have a `default_branch`, then the `#default_ref` method returns that value. The next thing it falls back to is the name of the default_ref attribute provided to the Environment object. You can see that accounted for in environment_spec.rb#29. The next thing it would fall back to is the hardcoded `master` in the config.rb file.